### ISSUE TITLE
shortcut for blacklisting whole package

### ISSUE THREAD

**Issue by hesxenon** [import-js/eslint-plugin-import#2532]
I think it'd be nice to have a shortcut identifier to blacklist a whole package from a given path.

like
```json
{
  "from": "domain/**/*",
  "target": "~contracts"
}
```
would prevent any file within `domain` from importing from the `contracts` package.

##### Note:
I feel like this should already be possible with the node resolver, but I'm not able to get it to work without dragging in lots of other errors...

**Comment by ljharb** [comment]
I'm a bit confused. Generally one only lints first-party code - or do you mean `./domain/**/*` the local files as opposed to the domain package?

Assuming that's what you mean, it seems like you can do this using the core https://eslint.org/docs/latest/rules/no-restricted-imports / https://eslint.org/docs/latest/rules/no-restricted-modules rules, and eslint overrides on that path?

**Comment by hesxenon** [comment]
the thing is that when using npm workspaces `contracts` _is_ first party code.

So imagine I have a frontend package with a hexagonal architecture; I'd like to be able to prevent direct imports from the contracts package _only_ in the domain layer. `no-restricted-modules` would blacklist that import everywhere in the frontend package, wouldn't it?

I can work around this by setting `"target": "../node_modules/contracts"` but that's a bit ugly :)

**Comment by hesxenon** [comment]
jeez, I see what you mean now... I've flipped "from" and "target"...

**Comment by ljharb** [comment]
in that case, you don't want to be linting the `contracts` that's inside node_modules - you want to be linting the actual workspace's location.

Using eslint overrides, you'd prevent that import only in the files you wanted to prevent it in.

**Comment by hesxenon** [comment]
no, I'm not linting the files inside `contracts`, I've flipped the "from" and "target" originally - sorry for the confusion. I want to prevent importing from `contracts` in `domain`.

I've adjusted the original text to reflect this

**Comment by ljharb** [comment]
OK - same solution applies tho.

Use eslint overrides to create a "rules" section (in the only eslintrc that should be in your repo, the one at the root) that only applies to files within `domain`. In that override, specify no-restricted-imports and no-restricted-modules so that files in `contracts` can not be required or imported.

**Comment by hesxenon** [comment]
yes, that works, same as directly pointing to the node_modules folder. However, I'd like to keep my rules for enforcing my architecture as close together as possible.

Is there a reasoning for not implementing the shortcut other than "there are existing solutions"? Don't get me wrong, nothing wrong with that reasoning, I just thought it'd be neat to have a shortcut.

**Comment by ljharb** [comment]
It’s added complexity that is already achievable. The slippery slope is that every rule has to add such a shortcut - why go down that road when overrides exist?
